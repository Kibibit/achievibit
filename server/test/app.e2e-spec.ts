import { INestApplication } from '@nestjs/common';
import { Test, TestingModule } from '@nestjs/testing';
import request from 'supertest';

import { AppModule } from '@kb-app';

describe('AppController (e2e)', () => {
  let app: INestApplication;
  let server;

  beforeEach(async () => {
    const moduleFixture: TestingModule = await Test.createTestingModule({
      imports: [AppModule]
    }).compile();

    app = moduleFixture.createNestApplication();
    await app.init();
    server = app.getHttpServer();
  });

  afterEach(async() => {
    await app.close();
  });

  test('/api (GET) API Information', async () => {
    const response = await request(server).get('/api');
    expect(response.status).toBe(200);
    expect(response.body).toMatchSnapshot();
  });

  test('/ (GET) HTML of client application', async () => {
    const response = await request(server).get('/');
    // console.log(response.body);
    expect(response.status).toBe(200);
    expect(response.type).toMatch(/html/);
  });
});
