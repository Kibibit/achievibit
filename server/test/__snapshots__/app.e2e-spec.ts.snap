// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppController (e2e) /api (GET) API Information 1`] = `
Object {
  "author": "thatkookooguy <neilkalman@gmail.com>",
  "bugs": "https://github.com/Kibibit/kb-server-client-template/issues",
  "description": "<DESCRIPTION>",
  "license": "MIT",
  "name": "@kibibit/kb-server-client-template",
  "repository": "https://github.com/Kibibit/kb-server-client-template.git",
  "version": "0.0.0-development",
}
`;
